agot-bg-game-server/src/common/ingame-game-state/action-game-state/resolve-march-order-game-state/ResolveMarchOrderGameState.ts

```diff
@@ -203,28 +162,18 @@ export default class ResolveMarchOrderGameState extends GameState<ActionGameStat
 
         if (adjacentCastleController) {
             // A castle with ships in port has been conquered
-            this.removePossibleOrdersInPort(portRegion);
-
             // return TakeControlOfEnemyPortGameState required
             return {
                 port: portRegion,
                 newController: adjacentCastleController
             }
         }
 
-        // When we reach this line the only way right now a orphaned ship can exist in a port
-        // is Martell playing Arianne in a non-capital city. So the ships have to be destroyed
-        const houseThatLostShips = portRegion.units.values[0].allegiance;
-
-        const destroyedShipCount = this.destroyAllShipsInPort(portRegion);
-        this.ingameGameState.log({
-            type: "ships-destroyed-by-empty-castle",
-            castle: adjacentCastle.name,
-            house: houseThatLostShips.name,
-            port: portRegion.name,
-            shipCount: destroyedShipCount
-        });
-
+        // We should never reach this line because we removed orphaned ships earlier.
+        // But as it doesn't hurt and to avoid a game crash we log a warning here and
+        // simply remove the orphaned ships again.
+        console.warn(`$Port with id '{portRegion.id}' contains orphaned ships which should have been removed before!`);
```

Comment:
I'd rather have a crash here. If a situation should never arise, it's better to know asap instead of trying to hot-fix.

If we leave it like this, when the situation arises, we will have no way of knowing as I don't have log monitoring atm. The users might not understand why some ships disappeared (or even not realize at all) and I'm not sure it's better that breaking their game.